include/xtensor/xgenerator.hpp

```diff
@@ -392,6 +397,13 @@ namespace xt
         return rebind_t<OR, OF>(std::move(func), shape_type(m_shape));
     }
 
+    template <class F, class R, class S>
+    template <class O>
+    inline auto xgenerator<F, R, S>::reshape(O&& shape) const
+    {
+        return reshape_view(*this, std::forward<dynamic_shape_type>(shape));
```

Comment:
... although that would mean more typing here.

Reply:
Would future overloads of reshape benefit from `dynamic_shape_type`? If it is really for just this one, I will put not use `using ...`.

Reply:
Maybe only the one accepting initializer_list if we decide to support the `reshape({-1, ...})` syntax. But we'll see later ;)